deps/rabbitmq_shovel_management/src/rabbit_shovel_mgmt.erl

```diff
@@ -107,52 +114,30 @@ filter_vhost_req(List, ReqData) ->
                            pget(vhost, I) =:= VHost]
     end.
 
-%% Allow users to see things in the vhosts they are authorised. But
-%% static shovels do not have a vhost, so only allow admins (not
-%% monitors) to see them.
-filter_vhost_user(List, _ReqData, #context{user = User = #user{tags = Tags}}) ->
-    VHosts = rabbit_mgmt_util:list_login_vhosts_names(User, undefined),
-    [I || I <- List, case pget(vhost, I) of
-                         undefined -> lists:member(administrator, Tags);
-                         VHost     -> lists:member(VHost, VHosts)
-                     end].
-
-status(ReqData, Context) ->
-    filter_vhost_user(
-      lists:append([status(Node) || Node <- [node() | nodes()]]),
-      ReqData, Context).
-
-status(Node) ->
-    case rpc:call(Node, rabbit_shovel_status, status, [], infinity) of
-        {badrpc, {'EXIT', _}} ->
-            [];
-        Status ->
-            [format(Node, I) || I <- Status]
+get_shovel_node(VHost, Name, ReqData, Context) ->
+    Shovels = rabbit_shovel_mgmt_util:status(ReqData, Context),
+    Match   = find_matching_shovel(VHost, Name, Shovels),
+    case Match of
+        undefined -> undefined;
+        Match     ->
+            {_, Node} = lists:keyfind(node, 1, Match),
+            Node
     end.
 
-format(Node, {Name, Type, Info, TS}) ->
-    [{node, Node}, {timestamp, format_ts(TS)}] ++
-        format_name(Type, Name) ++
-        format_info(Info).
-
-format_name(static,  Name)          -> [{name,  Name},
-                                        {type,  static}];
-format_name(dynamic, {VHost, Name}) -> [{name,  Name},
-                                        {vhost, VHost},
-                                        {type,  dynamic}].
-
-format_info(starting) ->
-    [{state, starting}];
-
-format_info({running, Props}) ->
-    [{state, running}] ++ Props;
-
-format_info({terminated, Reason}) ->
-    [{state,  terminated},
-     {reason, print("~p", [Reason])}].
-
-format_ts({{Y, M, D}, {H, Min, S}}) ->
-    print("~w-~2.2.0w-~2.2.0w ~w:~2.2.0w:~2.2.0w", [Y, M, D, H, Min, S]).
-
-print(Fmt, Val) ->
-    list_to_binary(io_lib:format(Fmt, Val)).
+%% This is similar to rabbit_shovel_status:find_matching_shovel/3
+%% but operates on a different input (a proplist of Shovel attributes)
+-spec find_matching_shovel(VHost :: vhost:name(),
+                           Name :: binary(),
+                           Shovels :: list()) -> 'undefined' | node().
+find_matching_shovel(VHost, Name, Shovels) ->
+    Matches = lists:filter(
```

Comment:
maybe `lists:search/2` would be more appropriate (and possibly address the dialyzer warning)